akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -51,6 +54,18 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
    */
   def lastStreamId(): Int = largestIncomingStreamId
 
+  private var maxConcurrentStreams = Http2Protocol.InitialMaxConcurrentStreams
+  def setMaxConcurrentStreams(newValue: Int): Unit = maxConcurrentStreams = newValue
+  /**
+   * @return true is the number of outgoing streams in Open (and HalfClosedXxx)
+   *         state doesn't exceed MaxConcurrentStreams
+   */
+  def hasOutgoingCapacity: Boolean = {
```

Comment:
Maybe something like that's a bit clearer?

```suggestion
  def hasCapacityToCreateStreams: Boolean = {
```